guardian/media-atom-maker
# Enable automatically generated ad breaks on YouTube

## What does this change?
In https://github.com/guardian/media-atom-maker/pull/953, we introduced changes to our video uploads by explicitly defining the ad format options applied to each video. In doing this, we may also have been unknowingly applying other options by default. It's not obvious from [YouTube's documentation](https://developers.google.com/youtube/partner/docs/v1/videoAdvertisingOptions#properties) whether these automatic overrides are happening, but users of MAM are reporting that videos which _should_ be eligible for midroll ads - because they are longer than 8 minutes - are not having them applied.

## How to test
In local or CODE, upload a new video clip that is longer than 8 minutes and publish it. Use the `.../api/youtube/video-info/<youtube-id-here>` endpoint in MAM to see which `adBreaks` are applied to the video; these should include midroll. Alternatively/additionally, navigate to the YouTube dashboard for the relevant channel and check the monetisation settings for the video.

## How can we measure success?
Midroll ads are automatically applied to all applicable videos (over 8 minutes long). YouTube channel managers can manage the placement of these ads via the channel's dashboard.

## Have we considered potential risks?


## Images
<img width="450" alt="Screenshot 2020-12-10 at 09 17 49" src="https://user-images.githubusercontent.com/12645938/101783729-9b66ae00-3ac8-11eb-9e26-2b8ef2622ef4.png">
